### ISSUE TITLE
[a11y]: Tab focus in accordion items incorrect, focusing on hidden items

### ISSUE THREAD

**Issue by KevinCamelo** [carbon-design-system/carbon#16781]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

@carbon/themes": "11.36.0

### React version

@carbon/react": "1.58.1

### Automated testing tool and ruleset

Manual testing

### Description

As the user tabs through a Carbon accordion component, the focus moves towards the items inside the accordion, even if the accordion is not expanded. 

If there are many focusable items embedded within an expanded accordion, this could result in the focus being lost for quite awhile until the user finds the next focusable item.

### WCAG 2.1 Violation

2.4.3 Focus Order

### Reproduction/example

https://stackblitz.com/edit/github-rtptpp-wsfdvm?file=src%2FApp.jsx

### Steps to reproduce

Tab through the accordion and notice how the focus gets unfocused as the user tabs through. As you tab, the focus moves to the "link" inside the accordion, even though the accordion is closed. 

**Video demo:**
![Recording](https://github.com/carbon-design-system/carbon/assets/29249866/aedc20f2-4763-4c87-a457-4cca40e35e16)

![tabs](https://github.com/carbon-design-system/carbon/assets/29249866/212c0c88-4ca7-431f-aa0e-67d78e42a95a)

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
This is a failure of [Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order)

I suspect it's due to the children only being hidden visually, not programmatically

**Comment by mbgower** [comment]
Yep, as per this screenshot, all that is happening is that somone is setting aria-expanded to "false". The child div is still in the dom and contains a tabbable element.
![Screenshot 2024-06-18 at 1 08 29 PM](https://github.com/carbon-design-system/carbon/assets/14298245/4a5b371b-cae5-409b-b9a7-b719600b38ac)
 Easily resolved by assigning "hidden" attribute on the wrapper div
`<div hidden="" class="cds--accordion__wrapper"><div id="accordion-item-:r1:" class="cds--accordion__content"><a href="#1">TEST</a></div></div>`
